test/gtest/uct/uct_test.h

```diff
@@ -284,6 +288,34 @@ class uct_test : public testing::TestWithParam<const resource*>,
         uct_iov_t               m_iov;
     };
 
+    struct async_event_ctx {
+        async_event_ctx() {
+            init();
+        }
+
+        ~async_event_ctx() {
+            reset();
+        }
+
+        void signal();
+        bool check_event(entity &e, int timeout);
+        void reset();
+
+        struct pollfd    wakeup_fd;
+        /* thios used for UCT TLs that support async event cb
+         * for event notification */
+        ucs_async_pipe_t aux_pipe;
+        bool             aux_pipe_init;
+    private:
+        void init() {
```

Comment:
why not just put it in ctor?

Reply:
we need `init()` to call from `reset()` method

Reply:
so why need both init and reset()? keep only reset() [and put its code in CC file]